coalib/settings/Setting.py

```diff
@@ -163,7 +163,11 @@ def __path__(self, origin=None, glob_escape_origin=False):
         :raises ValueError:        If no origin is specified in the setting
                                    nor the given origin parameter.
         """
-        strrep = str(self).strip()
+        if hasattr(self, 'value'):
+            strrep = self.value.strip()
```

Comment:
Because I think `self` will always have `value` and the `else` path will never be taken.